client/src/handlers/handlers.ts

```diff
@@ -14,6 +14,189 @@ export function makeHandleRunJob (
+    flixFileLocation = Filename
+}
+
+/**
+ * returns a count of total active terminals with prefix name `flix`.
+ * 
+ * @return number
+*/
+function countFlixTerminals() {
+    const activeTerminals = vscode.window.terminals
+    let count = 0
+    for (let i = 0; i < activeTerminals.length; i++) {
+        const element = activeTerminals[i];
+        if(element.name.substring(0, 4) == `flix`)
+            count+=1
+    }
+    return count
+}
+
+/**
+ * returns an active terminal with prefix name `flix`.
+ * 
+ * If not any active terminal with prefix name `flix`, it creates a new terminal with name `flix`.
+ *
+ * @return vscode.Terminal
+*/
+function ensureFlixTerminal() {
+    const activeTerminals = vscode.window.terminals
+    for (let i = 0; i < activeTerminals.length; i++) {
+        const element = activeTerminals[i];
+        if(element.name.substring(0, 4) == `flix`)
+            return element
+    }
+    return vscode.window.createTerminal(`flix`);
+}
+
+
+/**
+ * returns an new active terminal with prefix name `flix`.
+ * 
+ * If not any active terminal with prefix name `flix`, it creates a new terminal with name `flix` and returns it.
+ *
+ * If there are already `n` active terminals exist with prefix name `flix`, it creates a new terminal with name `flix n+1`
+ *
+ * @return vscode.Terminal
+*/
+function ensureNewFlixTerminal() {
+    const count = countFlixTerminals()
+    if(count == 0)
+        return vscode.window.createTerminal(`flix`);
+    else return vscode.window.createTerminal(`flix `+(count+1).toString())
+}
+
+/**
+ * takes a string and a terminal and passes that string to the terminal.
+ *
+ * @param cmd string (a terminal command) to pass to the terminal.
+ *
+ * @param terminal vscode.Terminal 
+ *
+ * @return void
+*/
+function passCommandToTerminal(cmd:string, terminal: vscode.Terminal) {
+    terminal.show();
+    terminal.sendText(cmd);
+}
+
+/**
+ * Opens an input box to ask the user for input.
+ * uses the `vscode.window.showInputBox` function with custom `prompt` and `placeHolder` and value of `ignoreFocusOut` to be `true`.
+ *
+ *
+ * @return A promise that resolves to a string the user provided or to `undefined` in case of dismissal.
+*/
+async function takeInputFromUser() {
+    const input = await vscode.window.showInputBox({
+        prompt: "Enter the space separated arguments",
+        placeHolder: "Foo bar",
+		ignoreFocusOut: true
+    })
+    return input
+}
+
+/**
+ * combines the paths of all flix files present in the current directory of vscode window.
+ * 
+ * gets an array of vscode.Uri for all flix files using `vscode.workspace.findFiles` function
+ *
+ * @return string of format "\<path_to_first_file\>" "\<path_to_second_file\>" ..........
+*/
+async function getFiles() {
+    const files = await vscode.workspace.findFiles(FLIX_GLOB_PATTERN)
+    let cmd = ""
+    for (let index = 0; index < files.length ; index++)
+        cmd += " \""+files[index].path+"\"";
```

Comment:
So this escaping is not really sufficient. At least in theory the filename could contain a " character. 

Maybe we should just use:

https://www.npmjs.com/package/shell-quote

and call that with an array of the right arguments.  I think that would work very well.